Library/src/Multiplayer/Components/ConversationSpaceComponent.cpp

```diff
@@ -205,17 +495,204 @@ void ConversationSpaceComponent::SetConversationInfo(const MessageInfo& Conversa
+						   "The retrieval of Conversation asset collections was not successful. ResCode: %d, HttpResCode: %d",
+						   (int) GetConversationResult.GetResultCode(),
+						   GetConversationResult.GetHttpResultCode());
+
+			const ConversationResult InternalResult(GetConversationResult.GetResultCode(), GetConversationResult.GetHttpResultCode());
+			INVOKE_IF_NOT_NULL(Callback, InternalResult);
+
+			return;
+		}
+
+		csp::systems::AssetCollectionResultCallback GetUpdatedConversationCallback
+			= [Callback, GetConversationResult, ConversationId, this](const csp::systems::AssetCollectionResult& GetUpdatedConversationResult)
+
+		{
+			if (GetUpdatedConversationResult.GetResultCode() == csp::systems::EResultCode::InProgress)
+			{
+				return;
+			}
+
+			if (GetUpdatedConversationResult.GetResultCode() == csp::systems::EResultCode::Failed)
+			{
+				CSP_LOG_FORMAT(csp::systems::LogLevel::Log,
+							   "The Update of Conversation asset collections was not successful. ResCode: %d, HttpResCode: %d",
+							   (int) GetConversationResult.GetResultCode(),
+							   GetConversationResult.GetHttpResultCode());
+
+				const ConversationResult InternalResult(GetConversationResult.GetResultCode(), GetConversationResult.GetHttpResultCode());
+				INVOKE_IF_NOT_NULL(Callback, InternalResult);
+
+				return;
+			}
+
+			const MultiplayerConnection::ErrorCodeCallbackHandler signalRCallback = [Callback, GetUpdatedConversationResult](ErrorCode Error)
+			{
+				if (Error != ErrorCode::None)
+				{
+					CSP_LOG_ERROR_MSG("SetConversationInfo: SignalR connection: Error");
+
+					INVOKE_IF_NOT_NULL(Callback, MakeInvalid<ConversationResult>());
+
+					return;
+				}
+
+				ConversationResult Result(GetUpdatedConversationResult.GetResultCode(), GetUpdatedConversationResult.GetHttpResultCode());
+				Result.FillConversationInfo(GetUpdatedConversationResult.GetAssetCollection());
+				INVOKE_IF_NOT_NULL(Callback, Result);
+			};
+
+            auto MultiPlayerConnection = csp::systems::SystemsManager::Get().GetMultiplayerConnection();
+			MultiPlayerConnection->SendNetworkEvent("ConversationSystem",
+													{ReplicatedValue((int64_t) ConversationMessageType::ConversationInformation), ConversationId},
+													signalRCallback);
+		};
+
+		MessageInfo NewConversationData(ConversationData);
+
+		if (NewConversationData.EditedTimestamp.IsEmpty())
+		{
+			const auto CurrentConversationData
+				= ConversationSystemHelpers::GetMessageInfoFromMessageAssetCollection(GetConversationResult.GetAssetCollection());
+			NewConversationData.EditedTimestamp = CurrentConversationData.EditedTimestamp;
+		}
+
+		AssetSystem->UpdateAssetCollectionMetadata(GetConversationResult.GetAssetCollection(),
+												   ConversationSystemHelpers::GenerateConversationAssetCollectionMetadata(NewConversationData),
+												   GetUpdatedConversationCallback);
+	};
+
+	AssetSystem->GetAssetCollectionById(ConversationId, GetConversationCallback);
```

Comment:
Do we need to make this call? We currently need it as the UpdateAssetCollectionMetadata method takes the AssetCollection object, but it only requires the Id. Can we not rather create an alternate method for updating metadata by Id or replace and deprecate the existing one? This feels inefficient as it stands.